src/event_server.cpp

```diff
@@ -2087,6 +2087,47 @@ static void get_calibration_data(JObj& response, const json_value *params)
     response << jrpc_result(rslt);
 }
 
+static void set_cooler_state(JObj& response, const json_value *params)
+{
+    Params p("enabled", params);
+    const json_value *val = p.param("enabled");
+    bool enable;
+    if (!val || !bool_param(val, &enable))
+    {
+        response << jrpc_error(JSONRPC_INVALID_PARAMS, "expected enabled boolean param");
+        return;
+    }
+
+    if (!pCamera || !pCamera->Connected)
+    {
+        response << jrpc_error(1, "camera not connected");
+        return;
+    }
+
+    if (!pCamera->HasCooler)
+    {
+        response << jrpc_error(1, "camera lacks a cooler");
+        return;
+    }
+
+    if (!pCamera->SetCoolerOn(enable))

```

Comment:
unfortunately the `Camera` and `Guider` APIs use the odd convention--predating my involvement in the project--returning `false` on success (!).  I'd love to change that at some point, but have never gotten around to it.
```suggestion
    bool error = pCamera->SetCoolerOn(enable);
    if (error)
```

Reply:
Would correcting this be in-scope for this change? I don't really think that anything else actually pays attention to the return values of the cooler funcs.

Reply:
You're right, other callers currently ignore the return value of SetCoolerOn.
I don't think that changing _just_ SetCoolerOn to have a different convention from the other APIs would be a good idea --  as it is now it may be an odd convention, but at least it is consistent through all the camera and guider APIs.  Having some APIs return false and others return true for success will make the code more fragile IMO.  My preference would be to not change the convention in this pr.  We can do a followup change at some point to fix it across all the APIs (or we can just leave it alone.)

Reply:
Alright. Because I was testing this new API method with a cooled QHY camera, I "fixed" the return boolean in the QHY driver improvements PR that was merged today, so I'll need to undo those and then reverse the logic in this one to match the, erm, convention ;) I have some feedback on camera functions from some folks so am working on a follow-up QHY PR right now, so I'll un-correct the return boolean in `SetCoolerOn()` and `SetCoolerSetpoint()` functions there as a part of that.